### PR TITLE
PRIME-1802 ODR: Move to a simpler architecture to manage volume

### PR DESCRIPTION
**Pull request by neophyte57**

### CODE REVIEW

**Review by AMiKaH — APPROVED** [review]

File: infrastructure/openshift/retrieve-pnet-logs.sh [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    echo -e "Last transaction id:  _${LAST_TX_ID}_\n"
+
+    UUID=$(cat /proc/sys/kernel/random/uuid)
+    echo -e "Generated request id:  _${UUID}_\n"
+
+    # CA certs need to be in place:  https://stackoverflow.com/questions/3160909/how-do-i-deal-with-certificates-using-curl-while-trying-to-access-an-https-url
+    # ls -l /etc/ssl/certs
+    curl --cert /opt/certs/prime-odr-api-cert.crt:${PRIME_ODR_API_SSL_CERT_PASSWORD} --key /opt/certs/prime-odr-api-cert.key --header "Authorization: Basic ${PRIME_ODR_API_ENCODED_CREDENTIALS}" \
+      "https://t1primedatasvc.maximusbc.ca/odr/prime/pnetdata/transactionLog?requestUUID=${UUID}&clientName=${PRIME_ODR_API_CLIENT_NAME}&lastTxnId=${LAST_TX_ID}&fetchSize=${PRIME_ODR_API_FETCH_SIZE}" | \
+      python3 /opt/scripts/parse_api_response.py | \
```
Comment (itchison) [inline infrastructure/openshift/retrieve-pnet-logs.sh:52]: Does this script prevent sql injection etc?

File: prime-dotnet-webapi/Migrations/20210901033644_PharmanetTransactionLogChangesAndDefaults.cs [hunk 2]
```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace Prime.Migrations
+{
+    public partial class PharmanetTransactionLogChangesAndDefaults : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.DropColumn(
```
Comment (itchison) [inline prime-dotnet-webapi/Migrations/20210901033644_PharmanetTransactionLogChangesAndDefaults.cs:10]: I think we should also just drop the audit columns. Maybe we need a creation timestamp but that's about it.

File: infrastructure/openshift/retrieve-pnet-logs.sh [hunk 3]
```diff
@@ -7,10 +7,14 @@ main() {
...
   echo -e "-------- STARTING CRON --------\n"
 
-  LAST_TX_ID=$(psql -h ${PGHOST} -d ${PGDATABASE} -U ${PGUSER} -W ${PGPASSWORD} -c 'select max(ptl."TransactionId") from "PharmanetTransactionLog" ptl')
+#  LAST_TX_ID=$(psql -h ${PGHOST} -d ${PGDATABASE} -U ${PGUSER} -W ${PGPASSWORD} -c 'select max(ptl."TransactionId") from "PharmanetTransactionLog" ptl')
+  LAST_TX_ID=$(psql -h ${PGHOST} -d ${PGDATABASE} -U ${PGUSER} -W ${PGPASSWORD} -c 'select count(*) from "Enrollee" e')
   echo -e ${LAST_TX_ID}
 
   ls -l /opt/certs
+
+  curl --cert /opt/certs/prime-odr-api-cert.crt:pwd4primetst --key /opt/certs/prime-odr-api-cert.key --header 'Authorization: Basic cHJpbWVkYXRhdXNlcjpWaWM5I1RzdDE=' -X GET https://t1primedatasvc.maximusbc.ca/odr/prime/pnetdata/transactionLog
```
Comment (itchison) [inline infrastructure/openshift/retrieve-pnet-logs.sh]: I see two secrets in this line of code. They are now compromised and we need to ask HIBC to rotate them.

**Review by itchison — COMMENTED** [review]
Please chat with me for the secrets.

File: infrastructure/openshift/retrieve-pnet-logs.sh (hunk 3 above)
Comment (neophyte57) [reply infrastructure/openshift/retrieve-pnet-logs.sh]: HIBC has completed updating 2021-09-08 and we are now using new secrets

File: prime-dotnet-webapi/Migrations/20210901033644_PharmanetTransactionLogChangesAndDefaults.cs (hunk 2 above)
Comment (neophyte57) [reply prime-dotnet-webapi/Migrations/20210901033644_PharmanetTransactionLogChangesAndDefaults.cs]: Done.

File: infrastructure/openshift/retrieve-pnet-logs.sh (hunk 1 above)
Comment (neophyte57) [reply infrastructure/openshift/retrieve-pnet-logs.sh:52]: The Postgres COPY command seems to be doing something smarter than simply concatenating input for an INSERT INTO statement (e.g. 
```
insert into "PharmanetTransactionLog"("TransactionId", "TxDateTime", "UserId", "SourceIpAddress", "LocationIpAddress", "PharmacyId", "TransactionType", "TransactionSubType", "PractitionerId", "CollegePrefix", "TransactionOutcome", "ProviderSoftwareId", "ProviderSoftwareVersion") 
values(16, TO_TIMESTAMP('20210726T185028', 'YYYYMMDDTHH24MISS'), '923456', '010.085.013.001', '010.085.013.193', 'BC00000J70', 'PE', '02', '700005', 'P1', 'TPN', '', ''); DELETE FROM "PlrProvider"; --')
```
)

Comment (neophyte57) [reply infrastructure/openshift/retrieve-pnet-logs.sh:52]: I tried the following:
```
echo "64,20210726T185028.52,923456,
010.085.013.001,010.085.013.193,BC00000J70,PE,02,700005,P1,TPN,,'); DELETE FROM "PlrProvider"; --" |  psql -h localhost -p 5432
-U postgres -c "\copy \"PharmanetTransactionLog\"(\"TransactionId\", \"TxDateTime\", \"UserId\", \"SourceIpAddress\", \"Location
IpAddress\", \"PharmacyId\", \"TransactionType\", \"TransactionSubType\", \"PractitionerId\", \"CollegePrefix\", \"TransactionOu
tcome\", \"ProviderSoftwareId\", \"ProviderSoftwareVersion\") FROM STDIN (FORMAT CSV)"
```

and nothing was deleted from the `PlrProvider` table and the following was inserted:

image

so the Postgres COPY command seems to mitigate SQL injection.

**Review by itchison — APPROVED** [review]
LGTM

**Review by mtpultz — APPROVED** [review]
There are some comments in retrieve-pnet-logs.cron.yml that could be removed if we're not using them, otherwise LGTM

**Comment by neophyte57** [comment]
> There are some comments in retrieve-pnet-logs.cron.yml that could be removed if we're not using them, otherwise LGTM

Cleaned up unused comments

File: infrastructure/openshift/retrieve-pnet-logs.sh (hunk 3 above)
Comment (neophyte57) [reply infrastructure/openshift/retrieve-pnet-logs.sh]: @itchison Please resolve conversation if appropriate

File: prime-dotnet-webapi/Migrations/20210901033644_PharmanetTransactionLogChangesAndDefaults.cs (hunk 2 above)
Comment (neophyte57) [reply prime-dotnet-webapi/Migrations/20210901033644_PharmanetTransactionLogChangesAndDefaults.cs]: @itchison Please resolve conversation if appropriate

**Review by i5okie — APPROVED** [review]

**Pull request merged**